### ISSUE TITLE
running verify unlocks all devices

### ISSUE THREAD

**Issue by superm1** [hughsie/fwupd#149]
I've found that running `#fwupdmgr verify` will unlock all devices that were locked and then not relock them when complete.

Furthermore due to the order of events, even devices that don't support checksums will be unlocked.

**Comment by hughsie** [comment]
So, Unlock() isn't meant to actually prevent them from being flashed (perhaps it's the wrong name) -- the unlock dance is done to avoid reading the PROM of PCI hardware unless it's really required. We found that unconditionally reading the PCI OptionROM on some hardware causes it to stop working or throw messages into the journal. I figured doing it as needed was better than unconditionally at startup, as the mitigation on broken hardware was "don't do verify-update" rather than "disable fwupd".
